community-tutorials/first-steps-to-protect-your-linux-server-against-common-attacks/01-en.md

```diff
@@ -0,0 +1,377 @@
+Then click **Generate**, and start moving the mouse within the Window. Putty uses mouse movements to collect randomness. The exact way you are going to move your mouse cannot be predicted by an external attacker. You may need to move the mouse for some time, depending on the size of your key. As you move it, the green progress bar should advance.
+
+Once the progress bar becomes full, the actual key generation computation takes place. This may take from several seconds to several minutes. When complete, the public key should appear in the Window. You can now specify a [passphrase](https://www.ssh.com/academy/ssh/passphrase) for the key.
+
+You should save at least the private key by clicking **Save private key**. It may be advisable to also save the public key, though it can be later regenerated by loading the private key (by clicking **Load**).
+
+![PuTTYgen generating RSA SSH key](images/PuTTYgen_generating_RSA_SSH_key-2.webp "PuTTYgen generating RSA SSH key")
+
+We strongly recommended using a passphrase be for private key files intended for interactive use. If keys are needed for automation (e.g., with [WinSCP](https://www.ssh.com/ssh/winscp/), then they may be left without a passphrase.
+
+![PutTTYgen created key and asking for passphase to save private key](images/puttygen_created_key_and_asking_for_passphase_to_save_private_key-2.webp "PutTTYgen created key and asking for passphase to save private key")
+
+## Step 3.2 - Use the Key-Pair
+
+### Step 3.2.1 - Add the Public Key
+
+Login to your remote server as your normal user ([See Step 2](#step-2---create-a-normal-user-account)) and add the public key to `~/.ssh/authorized_keys` so you can login to the SSH server with your private key.
+
+This can be done using the command line text editor *nano* for example:
+```bash
+sudo nano ~/.ssh/authorized_keys
+```
+1. Add the public key (starting with `ssh-ed25519 AAAA...` if you used ed25519) in a new line in the file
+2. Then save and close it by pressing `Ctrl + x` and `y` when it prompts "Save modified buffer?" on the bottom of the screen. Press `Enter` to save the file. 
+
+### Step 3.2.2 - Add the Private Key
+
+Then configure your SSH client to use your private key. 
+
+- If you connect to the remote server from the command line, the private keys in your home directory `~/.ssh` are automatically used.
+- With PuTTY first create a new session in the `Session` tab:
+  - In the upper field `Host Name` enter `yourusername@yourhost`
+
+    `yourhost` can be either an ip adress or the domain name of your remote server.
+  
+  - Port is 22 for now (default)
+  - In the field below `Saved Sessions` enter a name for your new session
+  - Set your private key at `Connection -> SSH -> Auth -> "Private key file for authentication"`
+  - Save your session by pressing `Save` in the `Session` tab.
+
+## Step 3.3 - Test the connection
+
+With the key pair now set up on the new created user account we can now test if the passwordless SSH connection works:
```

Comment:
My key gets refused. I think this has to do because i created the .ssh directory by myself.